server/cluster/cluster.go

```diff
@@ -919,7 +935,16 @@ func (c *RaftCluster) HandleStoreHeartbeat(heartbeat *pdpb.StoreHeartbeatRequest
 		peerInfo := core.NewPeerInfo(peer, loads, interval)
 		c.hotStat.CheckReadAsync(statistics.NewCheckPeerTask(peerInfo, region))
 	}
-
+	for _, stat := range stats.GetSnapshotStats() {
+		// the duration of snapshot is the sum between the send and generate snapshot.
+		// notice: to enlarge the limit in time, we reset the executing duration when it less than the minSnapshotDurationSec.
+		dur := stat.GetSendDurationSec() + stat.GetGenerateDurationSec()
+		if dur < minSnapshotDurationSec {
+			dur = minSnapshotDurationSec
+		}
+		e := int64(dur)*2 - int64(stat.GetTotalDurationSec())
```

Comment:
Please add some comments to explain how do we calculate the e?

Reply:
fixed